packages/numpy/patches/add-emscripten-cpu.patch

```diff
@@ -1,26 +1,26 @@
 diff --git a/numpy/core/include/numpy/npy_cpu.h b/numpy/core/include/numpy/npy_cpu.h
-index 84653ea18..ca25a5715 100644
+index 5edd8f42e..31560eac3 100644
 --- a/numpy/core/include/numpy/npy_cpu.h
 +++ b/numpy/core/include/numpy/npy_cpu.h
-@@ -82,6 +82,8 @@
-     #define NPY_CPU_ARCEL
- #elif defined(__arc__) && defined(__BIG_ENDIAN__)
+@@ -102,6 +102,8 @@
      #define NPY_CPU_ARCEB
+ #elif defined(__riscv) && defined(__riscv_xlen) && __riscv_xlen == 64
+     #define NPY_CPU_RISCV64
 +#elif defined(__EMSCRIPTEN__)
 +    #define NPY_CPU_EMSCRIPTEN
  #else
      #error Unknown CPU, please report this to numpy maintainers with \
      information about your platform (OS, CPU and compiler)
 diff --git a/numpy/core/include/numpy/npy_endian.h b/numpy/core/include/numpy/npy_endian.h
-index 1a42121db..5a43b14a8 100644
+index 649bdb0a6..ec2f6eda5 100644
 --- a/numpy/core/include/numpy/npy_endian.h
 +++ b/numpy/core/include/numpy/npy_endian.h
-@@ -46,7 +46,8 @@
-             || defined(NPY_CPU_SH_LE)   \
+@@ -47,7 +47,8 @@
              || defined(NPY_CPU_MIPSEL)  \
              || defined(NPY_CPU_PPC64LE) \
--            || defined(NPY_CPU_ARCEL)
-+            || defined(NPY_CPU_ARCEL)   \
+             || defined(NPY_CPU_ARCEL)   \
+-            || defined(NPY_CPU_RISCV64)
++            || defined(NPY_CPU_RISCV64) \
 +            || defined(NPY_CPU_EMSCRIPTEN)
```

Comment:
Do you think, something like this has a chance of being accepted into numpy upstream? It seems general enough.

Reply:
Possibly.  Might be worth a try.

(Some of the patches that were removed in this PR were because I already submitted them upstream...)